### DISCUSSION TITLE
Restricting GitHub App Permissions to Read-Only Access for Wiki Content

### DISCUSSION THREAD

**Discussion by Brakistad** [community/community#137305] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Question: Restricting GitHub App Permissions to Read-Only Access for Wiki Content

**Context:**

I’m working on setting up a GitHub App for my organization, and I’m specifically interested in allowing the app to read the content of a repository’s wiki. However, I want to restrict the app’s access strictly to the wiki content and not grant it access to the main source code of the repository.

**Question:**

Is there a way to configure a GitHub App’s permissions so that it has read-only access to a repository’s wiki without also granting read access to the repository’s source code? If so, how can this be achieved? If not, are there any recommended workarounds or best practices for securing source code while allowing wiki access?

Also, can i allow access to stricly "docs/*" folders in repos to allow for app access to image assets for the wikis. Also without any read access to other source code.

**Additional Details:**

- The repository in question is private, and the GitHub App is intended to automate tasks related to the documentation maintained in the wiki.
- I’ve reviewed the available permissions in the GitHub App settings but couldn’t find a way to separate wiki access from source code access.

Any guidance or suggestions would be greatly appreciated!

**Thanks in advance!**

**Comment by gabrielfrimodig** [comment, marked as answer]
From what I understand, GitHub's permission model doesn't allow for granular permissions specifically for wiki content without also giving access to the repository contents.

One potential workaround could be to use a separate repository just for documentation so that GitHub App can have read-only access to the documentation while isolating the main source code.

Alternatively, if the app needs access to specific files, you could use a GitHub Action to move those files to a location where the app can access them.

**Reply by Brakistad** [reply]
Yeah, the support is not there yet. And seperating the wiki from the repos is a acceptable workaround, altough it would be nice to avoid the clutter.
